### PR TITLE
Feature/stackn 31

### PR DESCRIPTION
**Pull request by stefanhellander**
STACKN-31
STACKN-79
STACKN-53
STACKN-79
STACKN-26 no longer relevant

Aside from STACKN-31 (this one turned out to need quite extensive changes):

Reworked how we authenticate to the API (now with Keycloak JWT tokens instead of the default drf tokens, see custom auth backend).
Added some CLI functionality  (stackn login, still needs work, functions for refreshing token, etc), started working on getting rid of project settings file, etc.

### CODE REVIEW

File: components/studio/models/views.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def index(request):
 
 @login_required(login_url='/accounts/login')
 def list(request, user, project):
+    print(dir(request))
```
Comment (dstoyanova) [inline components/studio/models/views.py:28]: Seems like a debug message that could be removed?

File: cli/scaleout/studioclient.py [hunk 2]
```diff
@@ -34,11 +37,14 @@ def __init__(self, config=None):
         self.password = self.config['password']
         # self.project_name = self.config['Project']['project_name']
         self.project_slug = self.config['Project']['project_slug']
-        self.auth_url = self.config['auth_url']
+        # self.auth_url = self.config['auth_url']
```
Comment (dstoyanova) [inline cli/scaleout/studioclient.py:40]: If we are not going to use this type of authentication anymore, maybe this should be removed?

File: cli/scaleout/studioclient.py [hunk 3]
```diff
@@ -34,11 +37,14 @@ def __init__(self, config=None):
...
         self.global_domain = self.config['so_domain_name']
         # TODO: This assumes a certain url-schema
-        self.api_url = self.auth_url.replace("/api-token-auth",'')
-        self.auth_headers = {'Authorization': 'Token {}'.format(self.config['token'])}
+        # self.api_url = self.auth_url.replace("/api-token-auth",'')
```
Comment (dstoyanova) [inline cli/scaleout/studioclient.py:43]: Same as above here.

File: cli/scaleout/cli/set_cmd.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+@main.group('set')
+@click.pass_context
+def set_cmd(ctx, daemon):
+    pass
+
+@set_cmd.command('remote')
+@click.pass_context
+def remote_cmd(ctx):
+    remote = input('Remote: ')
+    # os.
```
Comment (dstoyanova) [inline cli/scaleout/cli/set_cmd.py:24]: Seems like a typo?

File: cli/scaleout/studioclient.py (hunk 2 above)
Comment (stefanhellander) [reply cli/scaleout/studioclient.py:40]: I've removed this in the pull request for STACKN-103

File: cli/scaleout/studioclient.py (hunk 3 above)
Comment (stefanhellander) [reply cli/scaleout/studioclient.py:43]: I've removed this in the pull request for STACKN-103

File: cli/scaleout/cli/set_cmd.py (hunk 4 above)
Comment (stefanhellander) [reply cli/scaleout/cli/set_cmd.py:24]: It was an unfinished function, this is fixed in pull request STACKN-103

File: components/studio/models/views.py (hunk 1 above)
Comment (stefanhellander) [reply components/studio/models/views.py]: Yes, I've removed it now.

**Pull request merged**